src/ttsim/interface_dag_elements/fail_if.py

```diff
@@ -303,11 +304,13 @@ def input_data_is_invalid(input_data__flat: FlatData, xnp: ModuleType) -> None:
         )
         raise ValueError(message)
 
+    # Check that all arrays have the same length as the p_id array.
     len_p_id_array = len(input_data__flat[("p_id",)])
```

Comment:
```suggestion
    if not hasattr(input_data__flat[("p_id",)], "__len__"):
        raise ValueError("`p_id` must be an array or series.")
    len_p_id_array = len(input_data__flat[("p_id",)])
```
Else this will give "object of type int has no len" or so.